deducing-this.md

```diff
@@ -191,9 +190,9 @@ template <typename T>
 class optional {
     // ...
     template <typename Self>
-    decltype(auto) value(Self&& this self) {
+    decltype(auto) value(Self&& this) {
         if (o.has_value()) {
-            return std::forward<Self>(self).m_value;
+            return std::forward<std::like_t<Self, optional>>(*self).m_value;
```

Comment:
std::like_t isn't introduced until later.

Reply:
`*self` doesn't make sense - `self` is not a pointer.

We need `std::forward<std::like_t<Self, optional>>(*this).m_value` because otherwise, we might not be accessing the correct subobject. However, in this case, the optional is inferrable, and so my darling forward_like is a much better alternative:
```
std::forward_like<Self>(m_value)
```
This just transfers the qualifiers from Self to whatever the type of value is, no need for anything else.

Reply:
I added a section in 2. that introduces `forward_like` and `like` and links to @atomgalaxy 's implementation.